selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitPanelTest.java

```diff
@@ -0,0 +1,421 @@
+  private static final String NEW_PROJECT_NAME = "new-project";
+  private static final String RENAMED_JAVA_SPRING_GIT_PROJECT_NAME = "java-spring";
+
+  private static final String NODE_JS_EDITED_FILE_NAME = "app.js";
+  private static final String JAVA_SPRING_EDITED_FILE1_NAME = "AppController.java";
+  private static final String JAVA_SPRING_EDITED_FILE2_NAME = "web.xml";
+  private static final String JAVA_SPRING_DELETED_FILE_NAME = "spring-servlet.xml";
+  private static final String JAVA_SPRING_ADDED_FILE_NAME = "test.java";
+
+  @Inject
+  @Named("github.username")
+  private String gitHubUsername;
+
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private TestUser productUser;
+  @Inject private TestWorkspace workspace;
+  @Inject private Ide ide;
+  @Inject private Loader loader;
+  @Inject private TestProjectServiceClient projectServiceClient;
+  @Inject private Wizard wizard;
+  @Inject private Menu menu;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private CodenvyEditor editor;
+  @Inject private MachineTerminal terminal;
+  @Inject private AskDialog askDialog;
+  @Inject private AskForValueDialog askForValueDialog;
+  @Inject private TestUserPreferencesServiceClient testUserPreferencesServiceClient;
+  @Inject private org.eclipse.che.selenium.pageobject.git.Git git;
+  @Inject private GitCompare gitCompare;
+  @Inject private GitPanel gitPanel;
+
+  @BeforeClass
+  public void prepare() throws Exception {
+    testUserPreferencesServiceClient.addGitCommitter(gitHubUsername, productUser.getEmail());
+
+    URL resource;
+    resource = getClass().getResource("/projects/node-js-simple");
+    projectServiceClient.importProject(
+        workspace.getId(),
+        Paths.get(resource.toURI()),
+        NODE_JS_GIT_PROJECT_NAME,
+        ProjectTemplates.NODE_JS);
+
+    resource = getClass().getResource("/projects/ProjectWithDifferentTypeOfFiles");
+    projectServiceClient.importProject(
+        workspace.getId(),
+        Paths.get(resource.toURI()),
+        JAVA_SPRING_GIT_PROJECT_NAME,
+        ProjectTemplates.MAVEN_SPRING);
+
+    resource = getClass().getResource("/projects/plain-java-project");
+    projectServiceClient.importProject(
+        workspace.getId(),
+        Paths.get(resource.toURI()),
+        JAVA_PLAIN_NON_GIT_PROJECT_NAME,
+        ProjectTemplates.MAVEN_SPRING);
+
+    ide.open(workspace);
+
+    // make two projects to be under git
+    projectExplorer.waitProjectExplorer();
+    projectExplorer.waitItem(JAVA_PLAIN_NON_GIT_PROJECT_NAME);
+    projectExplorer.waitItem(NODE_JS_GIT_PROJECT_NAME);
+    projectExplorer.waitItem(JAVA_SPRING_GIT_PROJECT_NAME);
+
+    createGitRepositoryInProject(NODE_JS_GIT_PROJECT_NAME);
+    commitAllChangesInProject(NODE_JS_GIT_PROJECT_NAME, "initial commit");
+
+    createGitRepositoryInProject(JAVA_SPRING_GIT_PROJECT_NAME);
+    commitAllChangesInProject(JAVA_SPRING_GIT_PROJECT_NAME, "initial commit");
+  }
+
+  @Test(priority = 1)
```

Comment:
It's better to have independent tests inside the test class.

Reply:
Generally yes, that's better. But we have to take into account the cost of it.